api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +118,47 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCheckEnvironmentDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	cases := []string{"load_balancer", "service", "task"}
+	for _, c := range cases {
+		tagStore := tag.NewMemoryStore()
+		tags := []models.Tag{
+			models.Tag{
+				EntityID:   "env_id",
+				EntityType: "environment",
+				Key:        "name",
+				Value:      "env_name",
+			},
+			models.Tag{
+				EntityID:   "id",
+				EntityType: c,
+				Key:        "environment_id",
+				Value:      "env_id",
+			},
+		}
+
+		for _, tag := range tags {
+			if err := tagStore.Insert(tag); err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		target := provider.NewEnvironmentProvider(mockAWS.Client(), tagStore, mockConfig)
+		if err := target.Delete("env_id"); err != nil {
+			if serverError, ok := err.(*errors.ServerError); ok {
+				if serverError.Code != errors.DependencyError {
+					t.Errorf("Error: %s", err)
```

Comment:
If someone who didn't write this test saw this error message, I don't think they would know what is going on. I think we should have a message here explaining what went wrong vs. what was expected to happen.